core/base/mergeTreePrincipalGeodesics/MergeTreeAxesAlgorithmBase.h

```diff
@@ -0,0 +1,669 @@
+          persCorrelationMatrix[birthNode][g] = persCorr;
+          persCorrelationMatrix[deathNode][g] = persCorr;
+        }
+
+        // Push children to the queue
+        std::vector<ftm::idNode> children;
+        barycenter.tree.getChildren(node, children);
+        for(auto child : children)
+          queue.emplace(child);
+      }
+    }
+
+    //----------------------------------------------------------------------------
+    // Output Utils
+    //----------------------------------------------------------------------------
+    void zeroPadding(std::string &colName,
+                     const size_t numberCols,
+                     const size_t colIdx) {
+      std::string max{std::to_string(numberCols - 1)};
+      std::string cur{std::to_string(colIdx)};
+      std::string zer(max.size() - cur.size(), '0');
+      colName.append(zer).append(cur);
+    }
+
+    std::string getTableCoefficientName(int noGeodesics, int geodesicNum) {
+      std::string name{"T"};
+      zeroPadding(name, noGeodesics, geodesicNum);
+      return name;
+    }
+
+    std::string getTableCoefficientNormName(int noGeodesics, int geodesicNum) {
+      std::string name{"TNorm"};
+      zeroPadding(name, noGeodesics, geodesicNum);
+      return name;
+    }
+
+    std::string getTableVectorName(int noGeodesics,
+                                   int geodesicNum,
+                                   int vId,
+                                   int vComp,
+                                   bool isSecondInput = false) {
+      std::string indexString{};
+      zeroPadding(indexString, noGeodesics, geodesicNum);
+      std::string prefix{(isSecondInput ? "T2_" : "")};
+      std::string name{prefix + "V" + indexString + "_" + std::to_string(vId)
+                       + "_" + std::to_string(vComp)};
+      return name;
+    }
+
+    std::string getTableCorrelationName(int noGeodesics, int geodesicNum) {
+      std::string name{"Corr"};
+      zeroPadding(name, noGeodesics, geodesicNum);
+      return name;
+    }
+
+    std::string getTableCorrelationPersName(int noGeodesics, int geodesicNum) {
+      std::string name{"CorrPers"};
+      zeroPadding(name, noGeodesics, geodesicNum);
+      return name;
+    }
+
+    std::string getTableCorrelationTreeName(int noTrees, int treeNum) {
+      std::string name{"Tree"};
+      zeroPadding(name, noTrees, treeNum);
+      return name;
+    }
+
+    //----------------------------------------------------------------------------
+    // Testing
+    //----------------------------------------------------------------------------
+    void printVector(std::vector<double> &v, bool printHead = true) {
+      if(printHead)
+        std::cout << "======= printVector" << std::endl;
+      for(auto vv : v)
+        std::cout << vv << " ";
+      std::cout << std::endl;
+    }
+
+    void printVectorOfVector(std::vector<std::vector<double>> &v) {
+      std::cout << "======= printVectorOfVector" << std::endl;
```

Comment:
please remove `std::cout` and use `printMsg` instead

Reply:
Removed in the next commit!